client/a8c-for-agencies/components/onboarding-tour-modal/section-content.tsx

```diff
@@ -0,0 +1,30 @@
+import { ReactNode } from 'react';
+
+type Props = {
+	title: ReactNode;
+	descriptions?: ReactNode[];
+	hint?: ReactNode;
+};
+
+export default function OnboardingTourModalSectionContent( {
+	title,
+	descriptions = [],
+	hint,
+}: Props ) {
+	return (
+		<div className="onboarding-tour-modal__section-content">
+			<h1 className="onboarding-tour-modal__section-content-title">{ title }</h1>
+			<div className="onboarding-tour-modal__section-content-descriptions">
+				{ descriptions.map( ( description, index ) => (
+					<p
+						className="onboarding-tour-modal__section-content-description"
+						key={ `${ index }-${ description }` }
+					>
+						{ description }
+					</p>
+				) ) }
+			</div>
+			{ hint && <div className="onboarding-tour-modal__section-content-hint">{ hint }</div> }
+		</div>
+	);
+}
```

Comment:
I think it's better to keep the description as a single ReactNode instead of an array. If the description is complex JSX, converting it to a string could cause issues and might affect React's reconciliation

```suggestion
import { ReactNode } from 'react';

type Props = {
	title: ReactNode;
	descriptions?: ReactNode;
	hint?: ReactNode;
};

export default function OnboardingTourModalSectionContent( {
	title,
	descriptions,
	hint,
}: Props ) {
	return (
		<div className="onboarding-tour-modal__section-content">
			<h1 className="onboarding-tour-modal__section-content-title">{ title }</h1>
			<div className="onboarding-tour-modal__section-content-descriptions">
				{ description && (
				   <div className="onboarding-tour-modal__section-content-description">
					{ description }
				   </div>
			        ) }
			</div>
			{ hint && <div className="onboarding-tour-modal__section-content-hint">{ hint }</div> }
		</div>
	);
}
```

Reply:
Hi Yashwin,

I wanted to wrap the descriptions with `<p>` so we can achieve proper font style and spacing. Instead of altering this, I decided to change the descriptions to `(string | TranslateResult)[]` to limit the content and avoid complex JSX here. What do you think?